rust/rope/src/spans.rs

```diff
@@ -170,11 +171,10 @@ impl<T: Clone + Default> Spans<T> {
         let new_end = xform.transform(base_end, true);
         let mut builder = SpansBuilder::new(new_end - new_start);
         for (iv, data) in self.iter() {
-            let (start_closed, end_closed) = (iv.is_start_closed(), iv.is_end_closed());
-            let start = xform.transform(iv.start() + base_start, !start_closed) - new_start;
-            let end = xform.transform(iv.end() + base_start, end_closed) - new_start;
-            if start < end || (start_closed && end_closed) {
-                let iv = Interval::new(start, start_closed, end, end_closed);
+            let start = xform.transform(iv.start() + base_start, false) - new_start;
+            let end = xform.transform(iv.end() + base_start, true) - new_start;
```

Comment:
Oops yep that's an error.

Reply:
Agree after understanding this. I think the doc comment could be improved, perhaps from "at the given offset" to "into the region defined by base_start and base_end". And that could probably be an Interval rather than a pair of offsets, but maybe one thing at a time.